skimage/transform/registration.py

```diff
@@ -0,0 +1,135 @@
+    composed of the first 2 rows of the transformation matrix, as that is
+    all that is used in an affine transformation.
+
+    Parameters
+    ----------
+    parameter_vector : (N*(N+1)) array
+        Input array giving the argument of the minimum function to
+        optimise against
+
+    Returns
+    -------
+    matrix : (N+1, N+1) array
+        A transformation matrix used to obtain a new image
+    """
+
+    return np.concatenate(
+        (np.reshape(parameter_vector, (N, N + 1)), [[0] * N + [1]]), axis=0)
+
+
+def _matrix_to_parameter_vector(matrix):
+    """
+    Converts a 3x3 transformation matrix to the optimisation parameters
+
+    See the inverse function `_parameter_vector_to_matrix`.
+
+    Parameters
+    ----------
+    matrix : (N+1, N+1) array
+        A transformation matrix used to obtain a new image
+
+    Returns
+    -------
+    parameter_vector : (N*(N+1)) array
+        Output array giving the argument of the minimum function to
+        optimise against
+
+    """
+
+    return matrix[:-1, :].ravel()
+
+
+def register_affine(reference, target, *, cost=compare_mse, nlevels=None,
+                    iter_callback=lambda img, matrix: None):
+    """
+    Returns a matrix which registers the target image to the reference image
+
+    Will only return an affine transformation matrix.
+
+    Parameters
+    ----------
+    reference : ndimage
+        A reference image to compare against the target
+
+    target : ndimage
+        Our target for registration. Transforming this image using the
+        return value `matrix` aligns it with the reference.
+
+    cost : function, optional
+        A cost function which takes two images and returns a score which is
+        at a minimum when images are aligned. Uses the mean square error as
+        default.
+
+    nlevels : integer, optional
+        Change the maximum height we use for creating the Gaussian pyramid.
+        By default we take a guess based on the resolution of the image,
+        as extremely low resolution images may hinder registration.
+
+    iter_callback : function, optional
+        If given, this function is called once per pyramid level with the
+        current downsampled image and transformation matrix guess as the only
+        arguments. This is useful for debugging or for plotting intermediate
+        results during the iterative processs.
+
+    Returns
+    -------
+    matrix : array
+        A transformation matrix used to obtain a new image.
+        ndi.affine_transform(target, matrix) will align your target image.
+
+    Example
+    -------
+    >>> import numpy as np
+    >>> from skimage.data import camera
+    >>> reference_image = camera()
+    >>> r = 0.12
+    >>> c, s = np.cos(r), np.sin(r)
+    >>> matrix_transform = np.array([[c, -s, 0], [s, c, 50], [0, 0, 1]])
+    >>> target_image = ndi.affine_transform(reference_image, matrix_transform)
+    >>> output_matrix = register_affine(reference_image, target_image)
```

Comment:
Running register_affine produces a user warning. Can you update wherever register_affine calls _warps and explicitly specify the multichannel argument? Thanks!

```
scikit-image/skimage/transform/_warps.py:23: UserWarning: The default multichannel argument (None) is deprecated.  Please specify either True or False explicitly.  multichannel will default to False starting with release 0.16.
  warn('The default multichannel argument (None) is deprecated. 
```

Reply:
Should I create a separate PR for this then?

Reply:
@seanbudd that would be great!

Reply:
I merged in master and the warning appears to be gone. I'm guessing someone else fixed it at some point.

Reply:
wait nevermind I was wrong